LanguageFeatures/Patterns/type_inference_A29_t04.dart

```diff
@@ -52,7 +52,7 @@ String test2() {
 
 String test3() {
   if ({} as dynamic case <String, int>{"key1": String a}) {
```

Comment:
Wouldn't this be yet another case where there is no compile-time error? The context is refutable, and we're type checking `String a` with a matched value type of `int`, but that's OK in a refutable context.